src/ExpectedMark.js

```diff
@@ -62,31 +74,33 @@ export default class ExpectedMark extends UXBase {
 	}
 
 	record = () => {
-		if (
-			typeof window.performance !== 'undefined' &&
-			typeof window.performance.mark !== 'undefined'
-		) {
-			// record the mark using W3C User Timing API
-			window.performance.mark(this.props.name);
-		}
+		if (!this.marked) {
```

Comment:
never record same mark twice

Reply:
Does this solve our double-reporting issue? does it hide anything useful from user?

I am worried that you still call zone callbacks again even though you skip timeline marking.

Reply:
this will resolve the problem of a particular mark being set twice, yes. We want to always call zone callbacks because any zones that still have 'attached' listeners should know that a mark was set, even if it's an old mark. I _think_ it would be okay to _not_ call the zone callbacks and just assume that any zone that was interested in this mark would have 'heard' the first call, but I'm not 100% confident in that, and I think it's still safe to tell all listeners that the mark was recorded, even if the actual `mark.startTime` is in the past.

It will be useful to watch out for unexpected behavior related to this...